src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,117 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int MAXIMUM_CAPACITY = 1 << 30;
+
+    private int capacity;
+    private int threshold;
+    private float loadFactor;
+    private Node<K, V>[] buckets;
+    private int size;
+
+    public <K, V> MyHashMap() {
+        capacity = DEFAULT_INITIAL_CAPACITY;
+        loadFactor = DEFAULT_LOAD_FACTOR;
+        threshold = (int) (capacity * loadFactor);
+        buckets = new Node[capacity];
+        size = 0;
+    }
+
+    public <K, V> MyHashMap(int capacity) {
+        this.capacity = Math.min(capacity, MAXIMUM_CAPACITY);
+        loadFactor = DEFAULT_LOAD_FACTOR;
+        threshold = (int) (capacity * loadFactor);
+        buckets = new Node[capacity];
+        size = 0;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
+        if (buckets[index] == null) {
+            buckets[index] = new Node<>(key, value);
+        } else {
+            Node<K, V> prevNode = null;
+            Node<K, V> currentNode = buckets[index];
+            for (; currentNode != null;
+                    prevNode = currentNode, currentNode = currentNode.next) {
+                if (isKeysEquals(currentNode, key)) {
+                    currentNode.value = value;
+                    return;
+                }
+            }
+            prevNode.next = new Node<>(key, value);
+        }
+        size++;
+        checkResize();
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        for (Node<K, V> currentNode = buckets[index];
+                        currentNode != null;
+                        currentNode = currentNode.next) {
+            if (isKeysEquals(currentNode, key)) {
+                return currentNode.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private boolean isKeysEquals(Node<K, V> currentNode, K key) {
+        // currentNode must be != null
+        return ((key == currentNode.key) // including (key == currentNode.key == null)
+                || ((key != null)
+                && ((key.hashCode() == currentNode.hash)
+                && key.equals(currentNode.key))));
```

Comment:
```suggestion
        return ((key == currentNode.key) // including (key == currentNode.key == null)
                || ((key != null)
                && key.equals(currentNode.key))));
```

Reply:
I think, begining compare on hashcode will speedely. IF (key.hashCode() == currentNode.hash) is false, then more time-long (key.equals(currentNode.key)) will not execute.